share/p2p/peers/pool.go

```diff
@@ -221,3 +221,12 @@ func (p *pool) len() int {
 	defer p.m.RUnlock()
 	return p.activeCount
 }
+
+// reset will reset the pool to its initial state.
+func (p *pool) reset() {
+	lock := &p.m
+	lock.Lock()
+	defer lock.Lock()
+	// swap the pool with an empty one
+	*p = *newPool(time.Second)
+}
```

Comment:
So resetting works here by substituting existing pool with new pool. What if avoid allocating new pool by resetting each field on existing pool?

Reply:
Would be almost the same, because it would still need to realloc struct fields objects and maps. Probably would save 1 or 2 less pointer allocation. This reset operation happens once in 15 sec, so it is nothing for runtime. So I've optimised it for code reader performance 🙂